src/redux/features/Plantings/plantingsSelectors.ts

```diff
@@ -27,21 +29,39 @@ export const selectPlantingsDateRange = (state: RootState, dateRange: string[])
   }) ?? [];
 
 export const selectPlantingProgressSubzones = createSelector(
-  [(state: RootState) => selectPlantingSites(state), (state: RootState) => selectTotalPlantsBySubzone(state)],
-  (plantingSites, plantingsBySubzone) => {
-    plantingSites?.flatMap((ps) => {
-      return ps.plantingZones?.flatMap((zone) => {
-        return zone.plantingSubzones.map((sz) => {
-          return {
-            subzoneName: sz.fullName,
-            plantingComplete: sz.plantingCompleted,
-            plantingSite: ps.name,
-            zone: zone.name,
-            targetPlantingDensity: zone.targetPlantingDensity,
-            totalSeedlingsSent: plantingsBySubzone[sz.id].totalPlants,
-          };
-        });
+  [(state: RootState) => selectPlantingSites(state), (state: RootState) => selectPlantings(state)],
+  (plantingSites, plantings) => {
+    if (plantings) {
+      const plantingsBySubzone = getTotalPlantsBySubzone(plantings);
+      return plantingSites?.map((ps) => {
+        return {
+          siteId: ps.id,
+          siteName: ps.name,
+          reported: ps.plantingZones?.flatMap((zone) => {
+            return zone.plantingSubzones.map((sz) => {
+              return {
+                subzoneName: sz.fullName,
+                plantingComplete: sz.plantingCompleted,
+                plantingSite: ps.name,
+                zone: zone.name,
+                targetPlantingDensity: zone.targetPlantingDensity,
+                totalSeedlingsSent: plantingsBySubzone[sz.id].totalPlants,
+              };
+            });
+          }),
+        };
       });
-    });
+    }
   }
 );
+
+// selector for single site
+export const getReportedPlantsForSite = (state: RootState, siteId: number) =>
+  selectPlantingProgressSubzones(state)?.find((report) => report.siteId === siteId);
+
+// selector for search and filtering
```

Comment:
i think i made a mistake here. Not sure what the filters and searches will look like, we can remove this selector and add it later once we support search and filters?

Reply:
added a comment/question in figma and cc'd you

Reply:
We need it for searching by subzone and filtering by completed or not, so I will leave it

Reply:
I understand. This current selector does not filter by subzone or completed status. We can always add it later, that's easy to do.